### PR TITLE
Address bastion native client Issues and add proper error messages

### PR DESCRIPTION
**Pull request by Tejaswikandula**
This PR addresses the following bastion native client Issues
1. Add proper error message when --target-ip-address flag when IpConnect is disabled
2. Add support for premium SKU
3. Documentation fix and update preview flag
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by necusjz** [comment]
please record the test cases based on the latest azure-cli

**Review by Moazzem-Hossain-pixel — APPROVED** [review]
Thanks

File: src/bastion/azext_bastion/custom.py [hunk 1]
```diff
@@ -319,17 +317,30 @@ def rdp_bastion_host(cmd, target_resource_id, target_ip_address, resource_group_
...
-
+    if target_ip_address:
+        if 'enableIpConnect' in bastion and bastion['enableIpConnect'] is True:
+            return True
+        err_msg = "`--target-ip-address` flag cannot be used when IpConnect is not enabled. " \
+                  "Please use --target-resource-id flag instead."
+        raise InvalidArgumentValueError(err_msg)
     return False
 
+def _is_sku_in_allowed_skus(sku):
```
Comment (abhi7860) [inline src/bastion/azext_bastion/custom.py:328]: This is quite vague - please consider renaming

Comment (Tejaswikandula) [reply src/bastion/azext_bastion/custom.py]: Renamed to _is_sku_standard_or_higher

**Review by abhi7860 — APPROVED** [review]

**Comment by yonzhan** [comment]
Please fix CI issues

**Review by necusjz — APPROVED** [review]

**Pull request merged**